### PR TITLE
feat: Add Vulnerability delete action

### PR DESCRIPTION
**Pull request by chirino**
Task of https://github.com/trustification/trustify/issues/563

depends on https://github.com/trustification/trustify/pull/601

### CODE REVIEW

**Review by gildub — CHANGES_REQUESTED** [review]
Same issue here in regards of not using React Query layer for API requests.
Please update accordingly.

File: client/src/app/queries/vulnerabilities.ts [hunk 1]
```diff
@@ -46,6 +49,24 @@ export const useFetchVulnerabilityById = (id?: number | string) => {
...
 };
 
+export const useDeleteVulnerabilityByIdMutation = () => {
+  const queryClient = useQueryClient();
+  const {pushNotification} = React.useContext(NotificationsContext);
+
+  return useMutation({
+    mutationFn: (id: number | string) => deleteVulnerabilityById(id),
+    onSuccess: (_res, id) => {
+      queryClient.invalidateQueries({queryKey: [VulnerabilitiesQueryKey, id]});
```
Comment (gildub) [inline client/src/app/queries/vulnerabilities.ts:59]: The invalidateQueries key could actually be :
queryKey: [SBOMsQueryKey]

**Review by gildub — CHANGES_REQUESTED** [review]
LGTM just minor change, please see inline

**Review by carlosthe19916 — CHANGES_REQUESTED** [review]
Thanks for the PR. The same comments at https://github.com/trustification/trustify-ui/pull/120#pullrequestreview-2204769060 apply here

**Comment by chirino** [comment]
rolled into https://github.com/trustification/trustify-ui/pull/129

**Pull request closed without merging**